sympy/calculus/util.py

```diff
@@ -112,3 +119,662 @@ def elm_domain(expr, intrvl):
                                     for element in finite_set])
             _domain = Union(_domain, _domain_element)
         return _domain
+
+
+class Limits(AtomicExpr):
+    """
+    Limits represent an interval `[a, b]`, which is always closed at the
+    ends. Here `a` and `b` can be any value from extended real numbers.
+
+    The intended meaning of Limits is to give an approximate location of
+    the accumulation points of a real function at a limit point.
+
+    In many cases it would suffice to know that the limit set is bounded.
+    However, in some other cases more exact information could be useful.
+    For example, that all accumulation values of cos(x) + 1 are non-negative.
+    (Limits(-1, 1) + 1 = Limits(0, 2))
+
+    Let `a` and `b` be reals such that a <= b.
+
+    `\langle a, b\rangle = \{x \in \mathbb{R} \mid a \le x \le b\}`
+
+    `\langle -\infty, b\rangle = \{x \in \mathbb{R} \cup \{-\infty, \infty\} \mid x \le b\}`
```

Comment:
Here ``\infty`` should be outside (else it is not included).``-\infty`` could remain, but perhaps the result would look better if both were together at the end.